hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -414,71 +540,130 @@ Since the life cycle partition is the only partition that needs live updates in-
+
+Read and write commands transfer either 32bit or 64bit of data from the OTP to the corresponding CSR and vice versa. The access size is determined automatically, depending on whether the partition is scrambled or not. Also, (de)scrambling is performed transparently, depending on whether the partitions is scrambled or not.
+
+Digest calculation commands read out the complete contents of a particular partition, compute a digest and burn in that digest value to the predefined location at the end of the partition.
+
+Note that any unrecoverable OTP error will move the DAI into a terminal error state, where all access through the DAI will be locked.
+Also, the DAI consumes the read and write access information provided by the partition controller, and if a certain read or write access is not permitted, a recoverable error will be flagged in the status / error CSRs.
 
-#### Life-Cycle Partition
+### Life Cycle Interface Control
 
-**TODO: add FSM details**
+![Life Cycle Interface FSM](otp_ctrl_lci_fsm.svg)
+
+Upon reset release the LCI FSM waits until the OTP controller has initialized and the LCI gets enabled.
+Once it is in the idle state, incremental life cycle state updates can be initiated via the life cycle interface as [described here]({{< relref "#state-transitions" >}}).
+The LCI controller takes the life cycle state delta to be programmed and writes all non-blank 16bit words to OTP.
+In case of unrecoverable OTP errors, the FSM signals an error to the life cycle controller and moves into a terminal error state.
+
+### Key Derivation Interface
+
+![Key Derivation Interface FSM](otp_ctrl_kdi_fsm.svg)
+
+Upon reset release the KDI FSM waits until the OTP controller has initialized and the KDI gets enabled.
+Once it is in the idle state, key derivation and token hashing can be requested via the [token hashing]({{< relref "#token-hashing" >}}), [flash]({{< relref "#interface-to-flash-scrambler" >}}) and [sram]({{< relref "#interface-to-sram-and-otbn-scrambles" >}}) interfaces.
+Based on which interface makes the request, the KDI controller will evaluate a variant of the PRESENT digest mechanism as described in more detail below.
 
 ### Scrambling Datapath
 
-![OTP Digest Mechanism](otp_digest_mechanism.svg)
+![OTP Digest Mechanism](otp_ctrl_digest_mechanism.svg)
 
-The scrambling datapath is built around a single-round implementation of the [PRESENT lightweight cipher]({{< relref "hw/ip/prim/doc/prim_present" >}}) and contains some additional multiplexing circuitry to enable multiple modes of operation that are explained in more detail further below.
+The scrambling datapath is built around a single-round implementation of the [PRESENT lightweight cipher]({{< relref "hw/ip/prim/doc/prim_present" >}}) and contains some additional multiplexing circuitry to enable the DAI, KDI and partition controllers to evaluate different functions with the same datapath.
```

Comment:
I think "single-round" is really "unrolled single-cycle" right?  A single round of PRESENT doesn't do much ;)

Reply:
yeah that does not seem entirely right. it's not unrolled single-cycle though, since it's an iterative implementation that does one round per cycle. I'll amend that part.